### PR TITLE
Enable R tests in Jenkins

### PR DESCRIPTION
**Pull request by lxn2**
- Add test script to run Jenkins tests in Ubuntu
- Modify Ubuntu build slave environment to support R
- Modify existing R test script to be non-OSX specific

### CODE REVIEW

File: R-package/DESCRIPTION [hunk 1]
```diff
@@ -26,7 +26,6 @@ Suggests:
     knitr,
     rmarkdown,
     imager,
-    roxygen2
```
Comment (piiswrong) [inline R-package/DESCRIPTION]: ?

File: tests/travis/run_test.sh [hunk 2]
```diff
@@ -61,9 +61,13 @@ if [ ${TASK} == "r_test" ]; then
 
     set -e
     export _R_CHECK_TIMINGS_=0
-    wget https://cran.rstudio.com/bin/macosx/R-latest.pkg  -O /tmp/R-latest.pkg
-    sudo installer -pkg "/tmp/R-latest.pkg" -target /
-    Rscript -e "install.packages('devtools', repo = 'https://cran.rstudio.com')"
+    
+    if [[ ${TRAVIS_OS_NAME} == "osx" ]]; then
```
Comment (piiswrong) [inline tests/travis/run_test.sh]: why test for osx?

File: R-package/DESCRIPTION (hunk 1 above)
Comment (thirdwing) [inline R-package/DESCRIPTION]: This is due to the conflicts in this PR. The roxygen2 has removed by me last week. The latest version of roxygen2 broke the package building.

Comment (lxn2) [inline R-package/DESCRIPTION]: I'll rebase so this doesn't show up as my change.

File: tests/travis/run_test.sh (hunk 2 above)
Comment (lxn2) [inline tests/travis/run_test.sh]: Line 66 wget's a pkg for OSX, line 67 calls "installer" which seems to be an OSX tool? Has this test been running for linux at all?

Comment (thirdwing) [inline tests/travis/run_test.sh]: R travis testing is only enabled on OSX now. So it doesn't matter we test OSX or not.

Comment (lxn2) [inline tests/travis/run_test.sh]: Okay. I edited this because I want to run this script on a Jenkins test run too, recommended by @mli

**Comment by lxn2** [comment]
Fixed the merge conflict.

**Comment by lxn2** [comment]
run tests

**Comment by lxn2** [comment]
run tests

**Comment by lxn2** [comment]
run tests

**Pull request merged**